src/python/pants/reporting/report.py

```diff
@@ -72,6 +72,10 @@ def __init__(self):
     # We synchronize on this, to support parallel execution.
     self._lock = threading.Lock()
 
+  @property
+  def json_reporter(self):
```

Comment:
Adding knowledge of the _types_ of reporters to `Report` feels less than ideal, but the better option for getting the path to the `JsonReporter` output (another being constructing the path in the `RunTracker`, which felt more brittle, though there may be other options worth considering)

Reply:
Our comments look like we were worrying about the same thing. We spoke offline and I think agree this is less than ideal, but that the tangle that is reporting/report/run_tracker is too thick to unravel in this PR. I'm fine with calling the shot here in a TODO that references an issue tracking fixing the tangle. Perhaps adding an `assert` with a meaningful message in the meantime would be useful just in case a refactor finds us in the case where no json reporter is installed here.